packages/snjs/lib/services/api/api_service.ts

```diff
@@ -207,31 +212,80 @@ export class SNApiService extends PureService {
     }
   }
 
-  /**
-   * @param mfaKeyPath  The params path the server expects for authentication against
-   *                    a particular mfa challenge. A value of foo would mean the server
-   *                    would receive parameters as params['foo'] with value equal to mfaCode.
-   * @param mfaCode     The mfa challenge response value.
-   */
-  getAccountKeyParams(
+  private async getAuthMethods(
+    email: string,
+  ): Promise<HttpResponse<GetAuthMethodsResponse>> {
+    const params = this.params({ email });
+
+    return this.request({
+      verb: HttpVerb.Get,
+      url: joinPaths(this.host, Paths.v0.authMethods),
+      fallbackErrorMessage: messages.UNKNOWN_ERROR,
+      params,
+    }) as Promise<HttpResponse<GetAuthMethodsResponse>>
+  }
+
+  private async requestWithMfa(
+    httpRequest: HttpRequestWithFallbackError & {
+      identifier: string,
+    },
+  ): Promise<HttpResponse> {
+    const {identifier, ...restRequest} = httpRequest
+    for (;;) {
```

Comment:
What is this wizardry 🧐

Reply:
An infinite loop. Linter does not accept `while (true)`.

The loop was there previously, except implemented via recursion. It handles retrying if MFA code is incorrect.